apps/entry/serializers.py

```diff
@@ -685,12 +686,21 @@ def validate(self, data):
                     'image': f'You don\'t have permission to attach image: {image}',
                 })
         # If lead image is provided make sure lead are same
-        elif lead_image:
-            if lead_image.lead != lead:
+        elif lead_attachment:
+            if lead_attachment.lead != lead:
                 raise serializers.ValidationError({
-                    'lead_image': f'You don\'t have permission to attach lead image: {lead_image}',
+                    'lead_attachment': f'You don\'t have permission to attach lead attachment: {lead_attachment}',
                 })
-            data['image'] = lead_image.clone_as_deep_file(request.user)
+
+            if lead_attachment.type == LeadPreviewAttachment.AttachementFileType.XLSX:
+                data['entry_attachment'] = EntryAttachment.objects.create(
+                    file=lead_attachment.file,
```

Comment:
Let's clone the file as well. If LeadPreviewAttachment is deleted, we need to persist entry attachment data.